nemo/collections/tts/g2p/models/ja_jp_ipa.py

```diff
@@ -151,3 +156,191 @@ def __call__(self, text: str) -> List[str]:
+        self.ascii_letter_case = ascii_letter_case
+        # Load Japanese katakana grapheme set
+        ja_graphemes = GRAPHEME_CHARACTER_SETS.get("ja-JP", [])
+
+        pitch_markers = ['0', '1']
+        self.phoneme_list = sorted(list(ja_graphemes) + pitch_markers)
+
+        # ASCII letters handling
+        self.ascii_letter_dict = {
+            x: ascii_letter_prefix + x for x in get_grapheme_character_set(locale="en-US", case=ascii_letter_case)
+        }
+        self.ascii_letter_list = sorted(self.ascii_letter_dict)
+
+        self.punctuation = get_ipa_punctuation_list('ja-JP')
+
+        super().__init__(
+            word_tokenize_func=word_tokenize_func,
+            apply_to_oov_word=apply_to_oov_word,
+            mapping_file=mapping_file,
+        )
+
+    @staticmethod
+    def _split_katakana_to_moras(katakana: str) -> List[str]:
+        """Split Mora pattern: [main_katakana][small_katakana]? | [standalone_small] | [choonpu]"""
+        mora_pattern = r'[ア-ンヴ][ャュョァィゥェォヮ]?|[ァィゥェォヵヶッャュョヮ]|ー'
+        return re.findall(mora_pattern, katakana)
+
+    def _get_pitch_pattern(self, acc: int, total_mora: int) -> List[int]:
+        """Calculate pitch pattern for entire word chain.
+
+        Args:
+            acc: Accent nucleus position from first word in chain
+            total_mora: Total mora count of entire chain
+
+        Returns:
+            List of pitch values (0=low, 1=high) for each mora
+        """
+        if total_mora == 0:
+            return []
+
+        if acc == 0:  # Heiban: L-H-H-H...
+            return [0] + [1] * (total_mora - 1)
+
+        if acc == 1:  # Atamadaka: H-L-L-L...
+            return [1] + [0] * (total_mora - 1)
+
+        if acc >= total_mora:  # Odaka: L-H-H-H
+            return [0] + [1] * (total_mora - 1)
+
+        # Nakadaka: L-H...H-L...L (drop after acc-th mora)
+        return [0] + [1] * (acc - 1) + [0] * (total_mora - acc)
+
+    def _process_chain(self, chain: List[Dict], result: List[str]) -> None:
+        if not chain:
+            return
+
+        # Find chain starter
+        chain_starter_idx = 0
+        for i, word in enumerate(chain):
+            if word['chain_flag'] != 1:
+                chain_starter_idx = i
+                break
+
+        chain_acc = chain[chain_starter_idx]['acc']
+
+        # Split all words into moras
+        all_moras = []
+        for word in chain:
+            moras = self._split_katakana_to_moras(word['pron'])
+            all_moras.extend(moras)
+
+        # Calculate pitch pattern using chain starter's accent
+        total_mora = len(all_moras)
+        pitch_pattern = self._get_pitch_pattern(chain_acc, total_mora)
+
+        # Build output
+        for mora, pitch in zip(all_moras, pitch_pattern):
+            result.append(str(pitch))
+            result.extend(mora)
+
+    def __call__(self, text: str) -> List[str]:
+        """Convert Japanese text to kana with pitch accent markers."""
```

Comment:
could you add one example of input and output in this docstring? something like https://github.com/NVIDIA-NeMo/NeMo/blob/721ad1d3150011f691fab1604884ee595fb9a275/nemo/collections/tts/g2p/models/ja_jp_ipa.py#L135-L138